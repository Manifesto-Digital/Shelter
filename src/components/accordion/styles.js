import styled from 'styled-components';
import SVG from 'react-inlinesvg';
import { buttonReset } from '../styled/buttons';

export const Wrapper = styled.div`
  position: relative;

  &:not(:first-of-type) {
    &:before {
      position: absolute;
      content: ' ';
      left: 0;
      top: ${({ theme }) => `-${theme.spacing.small}`};
      width: 100%;
      border-top: ${({ theme, active }) =>
        active && `1px solid ${theme.palette.grey45}`};
    }
  }
`;

export const Heading = styled.h3`
  position: relative;
  font-weight: normal;
  color: ${({ theme }) => theme.palette.grey10};
  margin-bottom: ${({ active }) => active && '0'};
`;

export const HeadingButton = styled.button`
  position: relative;
  ${buttonReset};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.xl} ${theme.spacing.standard} ${
      theme.spacing.standard
    }`};
  width: 100%;
  text-align: left;
  font-size: ${({ theme }) => theme.headers.h3};
`;

export const Content = styled.div`
  display: ${({ isOpen }) => (isOpen ? 'block' : 'none')};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.standard} 0 ${
      theme.spacing.standard
    }`};
`;

export const ArrowSVG = styled(SVG)`
  position: absolute;
  transform: ${({ isOpen }) => isOpen && 'rotate(180deg)'} translateY(-50%);
  transform-origin: top;
  top: 50%;
  right: ${({ theme }) => theme.spacing.standard};
  width: 20px;
  height: 25px;

  svg {
    display: block;
    width: 20px;
    height: 25px;
  }
`;
