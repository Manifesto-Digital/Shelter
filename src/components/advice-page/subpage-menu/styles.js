import styled from 'styled-components';

export const MenuNav = styled.nav`
  margin-bottom: ${({ theme }) => theme.spacing.large};

  a {
    text-decoration: none;
  }
`;
