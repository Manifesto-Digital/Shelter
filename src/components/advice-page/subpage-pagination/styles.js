import styled from 'styled-components';
import SVG from 'react-inlinesvg';

export const StyledLink = styled.a`
  text-decoration: none;
  margin-right: ${({ type }) => (type === 'prev' ? 'auto' : null)};
  margin-left: ${({ type }) => (type === 'next' ? 'auto' : null)};

  span {
    text-decoration: underline;
    font-weight: normal;
  }
`;

export const DirectionIcon = styled.div`
  display: flex;
  flex-direction: ${({ type }) => (type === 'prev' ? 'row-reverse' : 'row')};
  justify-content: flex-end;
  font-size: ${({ theme }) => theme.fontsize.larger};
  margin: 0 0 0.4em;
`;

export const Wrapper = styled.div`
  display: flex;
  margin-bottom: ${({ theme }) => theme.spacing.large};
  padding-bottom: ${({ theme }) => theme.spacing.large};
  border-bottom: 2px solid ${({ theme }) => theme.palette.offWhite};
`;

export const StyledSVG = styled(SVG)`
  svg {
    display: block;
    width: 35px;
    height: 40px;
  }
`;
