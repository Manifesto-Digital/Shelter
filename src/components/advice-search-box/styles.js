import styled from 'styled-components';
import Button from '../button';
import { breakpoint } from '../theme/breakpoint';
import TextInput from '../forms/text-input';

export const AdviceSearchForm = styled.form`
  padding: ${props => props.theme.spacing.standard};
  background-color: ${props => props.theme.palette.offWhite};
`;

export const SearchInput = styled(TextInput)`
  margin-bottom: 1em;
  ${breakpoint.tablet`
        display: inline-block;
        width: 75%;
        margin-right: 2em;
    `};
`;

export const SearchButton = styled(Button)`
  background: ${props => props.theme.palette.sanMarinoBlue};
  position: relative;
  width: auto;
  padding: ${props => props.theme.spacing.small};
  color: ${props => props.theme.palette.white};
  display: block;
  ${breakpoint.tablet`
        display: inline-block;
    `};
`;
