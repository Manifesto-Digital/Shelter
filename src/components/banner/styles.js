import styled from 'styled-components'
import LinkHandler from '../link-handler'

export const Wrapper = styled.div`
  padding: ${props => props.theme.spacing.padding} 0;
  background-color: ${props =>
    (props.bg === 'red' && props.theme.palette.primary) ||
    (props.bg === 'blue' && props.theme.palette.sanMarino) ||
    (props.bg === 'black' && props.theme.palette.black) ||
    (props.bg === 'green' && props.theme.palette.donate)};
  margin-bottom: ${props =>
    props.removeMarginBottom ? '0' : props.theme.spacing.largePadding};
`

export const Header = styled.h2`
  color: ${props =>
    (props.bg === 'red' && props.theme.palette.white) ||
    (props.bg === 'blue' && props.theme.palette.white) ||
    (props.bg === 'black' && props.theme.palette.white) ||
    (props.bg === 'green' && props.theme.palette.white)};
`
export const StyledLinkHandler = styled(LinkHandler)`
  color: ${props =>
    (props.bg === 'red' && props.theme.palette.white) ||
    (props.bg === 'blue' && props.theme.palette.white) ||
    (props.bg === 'black' && props.theme.palette.white) ||
    (props.bg === 'green' && props.theme.palette.white)};
`
