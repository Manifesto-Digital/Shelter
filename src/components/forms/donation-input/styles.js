import styled from 'styled-components';
import { inputStyles } from '../../styled/inputs';

export const Wrapper = styled.span`
  position: relative;
`;

export const Input = styled.input`
  ${inputStyles}
  padding-left: 30px;
  max-width: ${props => (props.fullWidth ? '100%' : '10em')};

  &::-webkit-outer-spin-button,
  &::-webkit-inner-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }
`;

export const CurrencySymbol = styled.span`
  left: 12px;
  position: absolute;
  top: 50%;
  transform: translateY(-50%);
  z-index: 1;
  font-size: ${props => props.theme.fontsize.small};
`;
