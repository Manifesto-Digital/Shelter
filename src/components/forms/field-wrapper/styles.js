import styled from 'styled-components';

export const FormFieldWrapper = styled.div`
  margin: 0 0 0.8em;

  &:last-of-type {
    margin: 0;
  }
`;
