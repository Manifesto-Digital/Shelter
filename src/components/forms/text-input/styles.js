import styled, { css } from 'styled-components';
import { inputStyles } from '../../styled/inputs';
import successIcon from './success.png';
import errorIcon from './error.png';

const getStyles = ({ touched, error, theme }) => {
  if (!touched) {
    return css`
      border: 1px solid ${theme.palette.grey15};
      background-color: white;
    `;
  }

  if (error) {
    return css`
      border: 1px solid ${theme.palette.error};
      background: no-repeat 97% url(${errorIcon});
      background-color: ${theme.palette.errorLight};
    `;
  }

  return css`
    border: 1px solid ${theme.palette.success};
    background: no-repeat 97% url(${successIcon});
    background-color: ${theme.palette.successLight};
  `;
};

export const StyledInput = styled.input`
  ${getStyles};
  ${inputStyles}
`;
