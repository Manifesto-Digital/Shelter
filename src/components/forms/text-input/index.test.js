import React from 'react';
import TextInput from '.';
import {
  mountWithTheme,
  snapshotComponent,
} from '../../../../__tests__/helpers';
import { StyledInput } from './styles';

const createProps = (name, error, touched) => ({
  field: { name },
  form: {
    errors: { [name]: error },
    touched: { [name]: touched },
  },
});

['text', 'number', 'tel'].forEach(type => {
  it(`renders type as ${type}`, () => {
    const props = createProps('foo', undefined, false);

    const wrapper = mountWithTheme(<TextInput type={type} {...props} />);

    expect(wrapper.find(StyledInput).props().type).toBe(type);
  });
});

it('displays neutral state if not touched', () => {
  snapshotComponent(
    <TextInput {...createProps('foo', 'some error', false)} />,
    'with error'
  );
  snapshotComponent(
    <TextInput {...createProps('foo', undefined, false)} />,
    'no error'
  );
});

it('displays success if valid & touched', () => {
  const props = createProps('foo', undefined, true);

  snapshotComponent(<TextInput {...props} />);
});

it('displays error if invalid & touched', () => {
  const props = createProps('foo', 'Field is required', true);

  snapshotComponent(<TextInput {...props} />);
});

it('renders as textarea if type is textarea', () => {
  const props = createProps('foo', 'Field is required', true);

  const wrapper = mountWithTheme(<TextInput type="textarea" {...props} />);
  expect(wrapper.find(StyledInput).props().type).toBe('textarea');
});
