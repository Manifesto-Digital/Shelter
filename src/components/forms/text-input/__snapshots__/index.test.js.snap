// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays error if invalid & touched 1`] = `
.c0 {
  border: 1px solid #ea232d;
  background: no-repeat 97% url(test-file-stub);
  background-color: #ffe7ec;
  display: block;
  background-size: 15px 15px;
  outline: none;
  width: 100%;
  margin-bottom: 20px;
  padding: 0.65em;
}

<input
  className="c0"
  name="foo"
/>
`;

exports[`displays neutral state if not touched: no error 1`] = `
.c0 {
  border: 1px solid #d9d9d9;
  background-color: white;
  display: block;
  background-size: 15px 15px;
  outline: none;
  width: 100%;
  margin-bottom: 20px;
  padding: 0.65em;
}

<input
  className="c0"
  name="foo"
/>
`;

exports[`displays neutral state if not touched: with error 1`] = `
.c0 {
  border: 1px solid #d9d9d9;
  background-color: white;
  display: block;
  background-size: 15px 15px;
  outline: none;
  width: 100%;
  margin-bottom: 20px;
  padding: 0.65em;
}

<input
  className="c0"
  name="foo"
/>
`;

exports[`displays success if valid & touched 1`] = `
.c0 {
  border: 1px solid #62b33c;
  background: no-repeat 97% url(test-file-stub);
  background-color: #e6f8e7;
  display: block;
  background-size: 15px 15px;
  outline: none;
  width: 100%;
  margin-bottom: 20px;
  padding: 0.65em;
}

<input
  className="c0"
  name="foo"
/>
`;
